src/common-ssh/sftp.c

```diff
@@ -653,10 +654,19 @@ static int guac_common_ssh_sftp_ls_ack_handler(guac_user* user,
 
         /* Determine mimetype */
         const char* mimetype;
-        if (LIBSSH2_SFTP_S_ISDIR(attributes.permissions))
-            mimetype = GUAC_USER_STREAM_INDEX_MIMETYPE;
-        else
-            mimetype = "application/octet-stream";
+        //adding file size and permission
+        char tmpstr[150];
+	if (LIBSSH2_SFTP_S_ISDIR(attributes.permissions)) {
+		sprintf(tmpstr, "{\"mime\":\"%s\",\"size\":%llu,\"perm\":%lu}",
+			GUAC_USER_STREAM_INDEX_MIMETYPE, attributes.filesize,
+			attributes.permissions);
+	}
+	else {
+		sprintf(tmpstr, "{\"mime\":\"%s\",\"size\":%llu,\"perm\":%lu}",
+			"application/octet-stream", attributes.filesize,
+			attributes.permissions);
+	}
+	mimetype = tmpstr;
```

Comment:
I don't think we should just re-assign this string to `mimetype`, as the name `mimetype` implies that the data carried is the MIME type, not a string of information with various attributes about the file.

Reply:
+1

This would also break compatibility with any implementation that expects the value to be a mimetype string, which is also what we have defined for the protocol: https://guacamole.apache.org/doc/gug/protocol-reference.html#get-instruction

I think a better approach would be to leverage mimetype parameters to encode the information in a compatible way within the mimetype itself. This would allow the same information to be encoded for other instructions like `file`, as well. For example:

```
application/octet-stream;size=12345;mode=0644;uid=123;gid=456
```

If there are standard parameters for this, we can use them, or failing that we can define our own.

Reply:
Hello Mike,
I hope you're doing well. It's been some time since our last interaction, and I've taken this period to carefully consider and implement the MIME type encoding approach you recommended for the project.

With your guidance in mind, I've crafted a string that incorporates the additional file attributes as MIME type parameters, ensuring that the original distinction between files and directories is maintained. Here's the updated code snippet:

```c
char property_value[256];

snprintf(property_value, sizeof(property_value), 
         "%s;size=%llu;mode=%o;uid=%d;gid=%d", 
         mimetype, 
         (unsigned long long) attributes.filesize, 
         attributes.permissions & 07777,
         attributes.uid, 
         attributes.gid);

guac_common_json_write_property(user, stream, &list_state->json_state, absolute_path, property_value);
```

I would be grateful for your feedback on this revision. Does this change correctly reflect the format you envisioned? Your further input would be incredibly valuable.

Thank you for your guidance and patience.